libraries/standard/mqtt/src/mqtt.c

```diff
@@ -151,6 +169,61 @@ static MQTTStatus_t validateSubscribeUnsubscribeParams( const MQTTContext_t * co
 
 /*-----------------------------------------------------------*/
 
+static MQTTStatus_t receivePacket( MQTTContext_t * const pContext,
+                                   MQTTPacketInfo_t incomingPacket,
+                                   uint32_t remainingTimeMs )
+{
+    return MQTTSuccess;
+}
+
+/*-----------------------------------------------------------*/
+
+static MQTTStatus_t receiveConnack( MQTTContext_t * const pContext,
+                                    MQTTPacketInfo_t * const incomingPacket )
+{
+    MQTTStatus_t status = MQTTSuccess;
+    uint8_t retryCount = 0U;
+
+    assert( pContext != NULL );
+    assert( incomingPacket != NULL );
+
+    do
+    {
+        /* Transport read for incoming CONNACK packet type and length. */
+        status = MQTT_GetIncomingPacketTypeAndLength( pContext->transportInterface.recv,
```

Comment:
We must verify here that incoming packet is indeed a conn ack, before we try to read remaining packet.

Reply:
Agree.